docs/releases/4.0.txt

```diff
@@ -357,6 +357,18 @@ subdomains by setting :setting:`CSRF_COOKIE_DOMAIN` (or
 :setting:`SESSION_COOKIE_DOMAIN` if :setting:`CSRF_USE_SESSIONS` is enabled) to
 a value starting with a dot.
 
+``SecurityMiddleware`` no longer sets the ``X-XSS-Protection`` header
+---------------------------------------------------------------------
+
+The :class:`~django.middleware.security.SecurityMiddleware` no longer sets the
+the ``X-XSS-Protection`` header if the ``SECURE_BROWSER_XSS_FILTER`` setting is
+``True``. The setting is removed.
+
+Most modern browsers don't honor ``X-XSS-Protection`` HTTP header. If you want
+to set the header, use this line a custom middleware::
```

Comment:
I think that we should first recommend that a `Content-Security-Policy` that disables `'unsafe-inline'` should be used as documented [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection). Then we should say how to enable this if support for ancient browsers that do not support CSP must be maintained.

In short, this looks like we're taking away a security feature without explaining why or what it should be replaced with.

Reply:
I added a small note.

Reply:
@timgraham Do you want to take a look?

Reply:
Seems fine. The wording from Mozilla is "without allowing unsafe-inline scripts" (not sure if you intentionally dropped "scripts" here). Thanks.

Reply:
`scripts` added :+1: